### ISSUE TITLE
Material you color palette 

### ISSUE THREAD

**Issue by Fefedu973** [ohmae/orientation-faker#20]
Hi ! I love your app but I think you can add one functionality to make it even more integrated in the system ui. As you may know material you is the new Android interface style and I think it would be nice to add a "material you" color palette toggle to change the colors of the widget and the notification to the device accent color automatically.

**Comment by Christoph142** [comment]
This is one of my two small shortcomings of the app and why I rated it 4 / 5 in PlayStore. Bonus points, if it is possible to adapt the circle / squircle / rectangle corner roundings from the OS. 🤗

**Comment by ohmae** [comment]
In the next version, I plan to support Material3 Dynamic color (Material You). However, automatic adaptation is not yet possible; applying the current settings manually is the only option. Due to the complexity of coexisting with existing customization features, even if possible, it will take a bit more time. It's also possible to manually select from the Dynamic color palette. 
Next version is currently under review by Google. Recently, Google has been cautious about apps running in the background, which has led to longer review times.

||||
|--|--|--|
|![](https://github.com/ohmae/orientation-faker/assets/7691979/387b2ddb-a68e-48be-b9ee-aab668a14f76)|![](https://github.com/ohmae/orientation-faker/assets/7691979/2001fe83-bc51-46d7-9e54-16e00954b896)|![](https://github.com/ohmae/orientation-faker/assets/7691979/670952b7-0b7a-4917-998e-9a8cac0083ec)|

**Comment by Fefedu973** [comment]
> In the next version, I plan to support Material3 Dynamic color (Material You). However, automatic adaptation is not yet possible; applying the current settings manually is the only option. Due to the complexity of coexisting with existing customization features, even if possible, it will take a bit more time. It's also possible to manually select from the Dynamic color palette. 
> Next version is currently under review by Google. Recently, Google has been cautious about apps running in the background, which has led to longer review times.
> 
> ||||
> |--|--|--|
> |![](https://github.com/ohmae/orientation-faker/assets/7691979/387b2ddb-a68e-48be-b9ee-aab668a14f76)|![](https://github.com/ohmae/orientation-faker/assets/7691979/2001fe83-bc51-46d7-9e54-16e00954b896)|![](https://github.com/ohmae/orientation-faker/assets/7691979/670952b7-0b7a-4917-998e-9a8cac0083ec)|
> 

Nice !!

**Comment by ohmae** [comment]
Google's review process is over. I am now open to the public in open beta. 
If you sign up to be a beta tester, you can check it out. 😉

**Comment by Fefedu973** [comment]
Very nice update. I just have a question. Is it possible to switch the colors on theme change (light/dark) thanks 👍

**Comment by ohmae** [comment]
Good idea. I didn't have that perspective. And maybe the same method could be used to follow changes in dynamic color. I'll give it a try!

**Comment by ohmae** [comment]
I published new version 7.2.0 in open beta.  It now automatically follows dynamic color changes, including light/dark switching.

**Comment by Fefedu973** [comment]
Nice !

**Comment by Christoph142** [comment]
![Screenshot_2023-11-25-08-36-46-71_3038091bed58d514c8190464a747bb8e](https://github.com/ohmae/orientation-faker/assets/1440823/f6a600cc-503a-4753-923b-0f8948327a7b)
For some reason, the colors are different from what my system (Android 14) is displaying. Might be a ColorOs-specific issue, though.

**Comment by ohmae** [comment]
https://m3.material.io/styles/color/roles
Color roles are defined in Material3, but it is not clearly decided which one to use where. This app has the following settings.
- Selected: Primary/OnPrimary
- NonSelected: Surface/OnSurface
It's difficult to match on all devices because you don't know what colors the system UI will use.

**Comment by Christoph142** [comment]
Yeah, I figured as much. For some reason the color used by the UI is not part of the M3 palette at all. Not sure how that works.

**Comment by Fefedu973** [comment]
You can try fixing that using the repainter app
